### PR TITLE
backport config option to enable ClusterIP usage

### PR DESCRIPTION
**Pull request by DnPlas**
This is a backport from main to track/1.11 of the following:

* [Added change to istio-pilot to change ingress workload as ClusterIP](https://github.com/canonical/istio-operators/commit/7943735b67a56df05ae58d1d3c162b94347ab5fa)
* [code review comments applied.](https://github.com/canonical/istio-operators/commit/afb2a50c4a917b8dc758248505478eed12f6a840)
* [missing type key added](https://github.com/canonical/istio-operators/commit/9947502f59ae700940db9335aaa38a70edcdeb19)
* [Added ingress service type variable with options ClusterIP and Loadba…](https://github.com/canonical/istio-operators/commit/9c11f131ef110080569ebcb6c6120ccb4924c78e)

### CODE REVIEW

File: charms/istio-gateway/tests/unit/test_charm.py [hunk 1]
```diff
@@ -111,3 +111,30 @@ def test_removal(configured_harness, kind, mocked_client, mocker):
     mocked_client.return_value.delete.side_effect = api_error
     with pytest.raises(ApiError):
         configured_harness.charm.on.remove.emit()
+
+
+def test_service_type_cluserip(
+    configured_harness_only_ingress, gateway_service_type, mocked_client
```
Comment (i-chvets) [inline charms/istio-gateway/tests/unit/test_charm.py:117]: Will need docstring, otherwise linting will fail.

**Comment by i-chvets** [comment]
LGTM. Just CI needs to pass.

**Comment by DnPlas** [comment]
Hi @i-chvets, all changes necessary to make the CI pass are in #171. After we merge that PR, we can go ahead and rebase and merge this one.

**Review by ca-scribner — APPROVED** [review]

File: charms/istio-gateway/tests/unit/test_charm.py (hunk 1 above)
Comment (DnPlas) [reply charms/istio-gateway/tests/unit/test_charm.py:117]: This is not tested in the branch and has been already updated in more recent code.

**Pull request merged**